contract/contracts/NewInterfaces.sol

```diff
@@ -10,11 +17,30 @@ contract DistributerInterface {
         uint256 endDate
     );
 
-    string name;
-    address tokenHolder;
+    constructor(string memory _name, address _link) public {
+        name = _name;
+        link = _link;
+    }
+
+    string public name;
```

Comment:
`name` doesn't seem to have any particular use in the contract, but what about managing it as campaignInfoCid on the IPFS side?